toxcore/group.c

```diff
@@ -187,27 +240,51 @@ static int get_group_num(const Group_Chats *g_c, const uint8_t *identifier)
  * return peer number if peer is in chat.
  * return -1 if peer is not in chat.
  *
- * TODO(irungentoo): make this more efficient.
  */
-static int get_peer_index(Group_c *g, uint16_t peer_number)
+static aint get_peer_index(const Group_c *g, uint16_t peer_gid)
 {
-    uint32_t i;
+    aint i;
 
     for (i = 0; i < g->numpeers; ++i) {
-        if (g->group[i].peer_number == peer_number) {
+        if (g->peers[i].gid == (int)peer_gid) {
             return i;
         }
     }
 
     return -1;
 }
 
+static uint16_t find_new_peer_gid(const Group_c *g)
+{
+    uint16_t peer_number;
+    randombytes((uint8_t *)&peer_number, 2);
+
+    size_t tries = 0;
+
+    while (get_peer_index(g, peer_number) != -1) {
+        randombytes((uint8_t *)&peer_number, 2);
+        ++tries;
+
+        if (tries > 32) {
+            /* just find first unused peer number */
+            peer_number = 0;
```

Comment:
Would it make sense to start from the current random number, rather than starting from 0? What would be a reason against always doing that instead of trying 32 random numbers before resorting to linear search?